Can't build on Mac OS X -- function ASN1_FreeEncoded missing

Bisect says it was caused by 8e224d2f946e69f91216bae81b69bde2ffb495fd ; I can't find a definition for `ASN1_FreeEncoded' in the current source tree; I know the code has compiled for me since that commit, so I don't know what's up. But this happens when I pull the HEAD off git and try to compile.

```
cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_MANPAGES=OFF -DWITH_FFMPEG=ON -DWITH_CLIENT=OFF -DWITH_X11=OFF ..

[...]

Soyuz:~/ShortTerm/FreeRDP/build-lib$ make -j1

[...]
Undefined symbols for architecture x86_64:
  "_ASN1_FreeEncoded", referenced from:
      _ASN1_Encode in asn1.c.o
ld: symbol(s) not found for architecture x86_64
collect2: ld returned 1 exit status
make[2]: *** [winpr/asn1/libwinpr-asn1.1.0.1.dylib] Error 1
make[1]: *** [winpr/asn1/CMakeFiles/winpr-asn1.dir/all] Error 2
make: *** [all] Error 2
i686-apple-darwin11-llvm-gcc-4.2 (GCC) 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.9.00)
```

Build completes on my Ubuntu machine.

```
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,10 +108,18 @@ endif()
 
 # Mac OS X
 if(APPLE)
-	include_directories(/opt/local/include)
-	link_directories(/opt/local/lib)
-	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -mmacosx-version-min=10.4")
-	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -framework CoreFoundation")
+	if(IS_DIRECTORY /opt/local/include)
+		include_directories(/opt/local/include)
+		link_directories(/opt/local/lib)
+	endif()
+	
+	option(WITH_CLANG "Build using clang" OFF)
+	if(WITH_CLANG)
+		set(CMAKE_C_COMPILER "clang")
+	endif()
+	
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mmacosx-version-min=10.4")
+	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl-framework,CoreFoundation")	
 endif()
 
 if(NOT WIN32)

--- a/winpr/CMakeLists.txt
+++ b/winpr/CMakeLists.txt
@@ -17,6 +17,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if (APPLE)
+	# flat_namespace should be avoided, but is required for -undefined warning. Since WinPR currently has
+	# a lot of undefined symbols in use, use this hack until they're filled out.
+	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-flat_namespace,-undefined,warning")	
+endif()
+
 add_subdirectory(crt)
 add_subdirectory(utils)
 add_subdirectory(heap)

